pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/instance/MailingAddressComponent.java

```diff
@@ -1,42 +1,77 @@
 package org.broadinstitute.ddp.model.activity.instance;
 
+import java.util.NoSuchElementException;
 import java.util.function.Consumer;
 
 import com.google.gson.annotations.SerializedName;
-
 import org.broadinstitute.ddp.content.ContentStyle;
+import org.broadinstitute.ddp.content.HtmlConverter;
 import org.broadinstitute.ddp.model.activity.types.ComponentType;
 
 public class MailingAddressComponent extends FormComponent {
 
     @SerializedName("parameters")
     private SerializedFields serializedFields;
 
-    public MailingAddressComponent(String titleText, String subtitleText, boolean hideNumber) {
+    private transient Long titleTemplateId;
+    private transient Long subtitleTemplateId;
+
+    public MailingAddressComponent(Long titleTemplateId, Long subtitleTemplateId, boolean hideNumber) {
         super(ComponentType.MAILING_ADDRESS);
         hideDisplayNumber = hideNumber;
-        serializedFields = new SerializedFields(titleText, subtitleText);
-    }
-
-    public static class SerializedFields {
-        @SerializedName("titleText")
-        private static String titleText;
-        @SerializedName("subtitleText")
-        private static String subtitleText;
-
-        public SerializedFields(String titleText, String subtitleText) {
-            this.subtitleText = subtitleText;
-            this.titleText = titleText;
-        }
+        titleTemplateId = titleTemplateId;
+        subtitleTemplateId = subtitleTemplateId;
+        serializedFields = new SerializedFields();
     }
 
     @Override
     public void registerTemplateIds(Consumer<Long> registry) {
-        // Implement this method once we decide to use bulk rendering for this component
+        if (titleTemplateId != null) {
+            registry.accept(titleTemplateId);
+        }
+        if (subtitleTemplateId != null) {
+            registry.accept(subtitleTemplateId);
+        }
     }
 
     @Override
     public void applyRenderedTemplates(Provider<String> rendered, ContentStyle style) {
-        // Implement this method once we decide to use bulk rendering for this component
```

Comment:
Updated mailing-address component so it hooks into the bulk rendering process and can take advantage of those special vars.